### PR TITLE
CORE-742: Improve BTC transaction state handling in crypto

### PR DESCRIPTION
**Pull request by michael-brd**
@EBGToo, placing this in DRAFT for now as I test further but wanted early eyes on it and the approach. In API testing, things look good. Just waiting for a sync to confirm P2P behaviour.

### CODE REVIEW

**Comment by michael-brd** [comment]
@EBGToo, changed to "ready" as this looks good in testing.

File: bitcoin/BRWalletManager.c [hunk 1]
```diff
@@ -412,6 +414,39 @@ BRWalletSweeperValidate (BRWalletSweeper sweeper) {
  * The key to this design is that the `refedTransaction` is NEVER, EVER dereferenced. It is
  * used solely for lookups based on identify in callbacks from the BRWallet and
  * BRPeerManager (via the BRSyncManager).
+ *
+ *
+ * A Word About State
+ *
```
Comment (EBGToo) [inline bitcoin/BRWalletManager.c]: Nice explanation.

File: crypto/BRCryptoWalletManagerClient.c [hunk 2]
```diff
@@ -698,7 +666,7 @@ cwmTransactionEventAsBTC (BRWalletManagerClientContext context,
 
             BRCryptoTransferState oldState = cryptoTransferGetState (transfer);
 
-            if (CRYPTO_TRANSFER_STATE_INCLUDED == oldState.type &&
+            if (CRYPTO_TRANSFER_STATE_SUBMITTED != oldState.type &&
                 (0 == event.u.updated.timestamp || TX_UNCONFIRMED == event.u.updated.blockHeight)) {
                 // The transfer is not included so set it to the submitted state at this point
                 BRCryptoTransferState newState = cryptoTransferStateInit (CRYPTO_TRANSFER_STATE_SUBMITTED);
```
Comment (EBGToo) [inline crypto/BRCryptoWalletManagerClient.c]: Might be the comment 'The transfer is not included ...' is inconsistent with the logic.  Maybe a general explanation in front of the `if` - as in "if the transfer is not submitted and is missing values for timestamp and blockHeight, then return it to submitted" ?

File: crypto/BRCryptoWalletManagerClient.c [hunk 3]
```diff
@@ -652,38 +652,6 @@ cwmTransactionEventAsBTC (BRWalletManagerClientContext context,
                 cryptoUnitGive (unit);
            }
 
-            // ... update state to reflect included if the timestamp and block height are already set
-            if (0 != btcTransaction->timestamp && TX_UNCONFIRMED != btcTransaction->blockHeight) {
-
```
Comment (EBGToo) [inline crypto/BRCryptoWalletManagerClient.c]: Maybe a comment in the 'empty code' to the effect: "We won't announce a state change because ..."

**Pull request merged**